File: kotlin/Prime.kt
object Prime {
  internal class Sieve(prime:Int) {
    private val next:Sieve
    private val prime:Int = 0

    init {
      this.prime = prime
    }

    fun test(n:Int) {Î
      if (this.prime * (n / this.prime) == n)
      {
      }
      else if (this.next == null)
      {
        println(n)
        this.next = Sieve(n)
      }
      else
      {
        this.next.test(n)
      }
    }
  }
  
  @JvmStatic fun main(args:Array<String>) {
    val s = Sieve(2)
    val i = 2
    println("2")
    while (i < 13)
    {
      i += 1
      s.test(i)
    }
  }
}

File: csharp/Prime.cs
using System;

internal class Sieve
{
    private readonly int _prime;

    private Sieve _next;
    
    public Sieve(int prime)
    {
        _prime = prime;
    }
    
    public void Test(int n)
    {
        if (_prime * (n / _prime) == n)
        {

        }
        else if (_next == null)
        {

            Console.WriteLine(n);
            _next = new Sieve(n);
        }
        else
        {
            _next.Test(n);
        }
    }
}

public class Prime
{
    public static void Main(string[] args)
    {
        var s = new Sieve(2);
        var i = 2;

        Console.WriteLine("2");

        while (i < 10)
        {
            i += 1;
            s.Test(i);
        }
    }
}
